### PR TITLE
test(datastore): custom pk tests

### PR DESCRIPTION
**Pull request by diegocstn**
*Description of changes:*
DataStore Custom Primary Key tests

*Check points: (check or cross out if not relevant)*

~- [ ] Added new tests to cover change, if needed~
- [ ] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [ ] All integration tests pass
~- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)~
~- [ ] Documentation update for the change if required~
- [x] PR title conforms to conventional commit style
~- [ ] If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Storage/CascadeDeleteOperationTests.swift [hunk 1]
```diff
@@ -90,6 +92,47 @@ class CascadeDeleteOperationTests: StorageEngineTestsBase {
...
+            return
+        }
+        let predicate: QueryPredicate = ModelCompositePk.keys.id == modelId
+            && ModelCompositePk.keys.dob == modelDob
+        guard case .success(let queriedRestaurants) = queryModelSynchronous(modelType: ModelCompositePk.self,
+                                                                            predicate: predicate) else {
+            XCTFail("Failed to query")
+            return
+        }
+        XCTAssertEqual(queriedRestaurants.count, 1)
```
Comment (thisisabhash) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Storage/CascadeDeleteOperationTests.swift:110]: nit: can add assertions on the fields of `model`

File: AmplifyTestCommon/Helpers/AuthSignInHelper.swift [hunk 2]
```diff
@@ -11,12 +11,12 @@ import XCTest
 public typealias CompletionType = (Bool, AuthError?) -> Void
 
 public struct AuthSignInHelper {
-
```
Comment (thisisabhash) [inline AmplifyTestCommon/Helpers/AuthSignInHelper.swift:14]: This file can be reverted.

**Review by thisisabhash — APPROVED** [review]
LGTM, with minor comments

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Storage/CascadeDeleteOperationTests.swift (hunk 1 above)
Comment (diegocstn) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Storage/CascadeDeleteOperationTests.swift:110]: good call, updated with assertions and renamed to `querieModel`

**Pull request merged**